text/0000-afp.md

```diff
@@ -0,0 +1,354 @@
+# ARA FARMING PROTOCOL
+
+---
+Request For Comments: 0000
+---
+Author(s): Brandon Plaster, Huy Dao
+---
+Category: Distributed Services
+---
+Date: June, 2018
+---
+
+## 0. Abstract
+
+Ara Farming Protocol (AFP) provides methods for distributed decentralized services, or decents, in the Ara Network to request work from its peers and determine the specifications of work contracts (i.e. rewards and other service-specific details).
+
+## 1. Status
+
+This RFC is under _active development and consideration_.
+
+## 2. Introduction
+
+AFP defines a set of extensible classes in Javascript which enable peers of a distributed service to communicate about and define a statement of work for that service. Peers exchange messages in the form of [farming-protocol-buffers](https://github.com/AraBlocks/farming-protocol-buffers).
+
+AFP also provides a default implementation for communicating over duplex streams.
+
+A [farmer](#412-farmer) would extend the AFP Farmer class to define that farmer’s specifications for generating a quote for a task, validating a peer/SOW for a task, and signing and validating an agreement for a task. The farmer could then use a discovery-swarm (or some other peer discovery method) to broadcast their ability to complete a task, and then communicate via a duplex stream their capabilities and costs.
```

Comment:
`SOW` should link to the definition below, as farmer and requester do (or maybe even in the first reference to `statement of work` in line 23.  Also, it is sometimes referred to as `statement-of-work` `SOW` and `statement of work`, maybe it should be standardized/consistent.

Reply:
Will update and make consistent. Question, should every reference to it be a link, or just the initial ones?